src/Physics/ini_model_DR.f90

```diff
@@ -363,6 +381,50 @@ SUBROUTINE rotateStressToFaultCS(EQN,MESH,nBndGP,s_xx,s_yy,s_zz,s_xy,s_yz,s_xz,s
     enddo
   END SUBROUTINE rotateStressToFaultCS
 
+  SUBROUTINE rotateSlipToFaultCS(EQN, MESH, nBndGP, StrikeSlip , DipSlip, SlipInFaultCS)
+    USE common_operators_mod
+    !-------------------------------------------------------------------------!
+    IMPLICIT NONE
+    !-------------------------------------------------------------------------!
+    TYPE(tEquations)                      :: EQN
+    TYPE(tUnstructMesh)                   :: MESH
+    integer                               :: nBndGP
+    real, allocatable, dimension(:,:)     :: StrikeSlip, DipSlip
+    real, allocatable, dimension(:,:,:)   :: SlipInFaultCS
+    !-------------------------------------------------------------------------!
+    ! Local variable declaration
+    integer                             :: i
+    real                                :: NormalVect_n(3), NormalVect_s(3), NormalVect_t(3)
+    real                                :: strike_vector(3), dip_vector(3),crossprod(3)
+    real                                :: scalarprod, cos1, sin1
+    !-------------------------------------------------------------------------!
+    intent(in)                          :: EQN,MESH,nBndGP
+    intent(inout)                       :: StrikeSlip, DipSlip
+    intent(inout)                       :: SlipInFaultCS
+    do i = 1, MESH%Fault%nSide
+          NormalVect_n = MESH%Fault%geoNormals(1:3,i)
+          NormalVect_s = MESH%Fault%geoTangent1(1:3,i)
+          NormalVect_t = MESH%Fault%geoTangent2(1:3,i)
+ 
+          strike_vector(1) = NormalVect_n(2)/sqrt(NormalVect_n(1)**2+NormalVect_n(2)**2)
+          strike_vector(2) = -NormalVect_n(1)/sqrt(NormalVect_n(1)**2+NormalVect_n(2)**2)
+          strike_vector(3) = 0.0D0
+          dip_vector = NormalVect_n .x. strike_vector
+          dip_vector = dip_vector / sqrt(dip_vector(1)**2+dip_vector(2)**2+dip_vector(3)**2)
```

Comment:
This is a duplicate from create_fault_rotationmatrix. I'd introduce another subroutine in create_fault_rotationmatrix.f90 which returns these three vectors, which is called from here and from create_fault_rotationmatrix

Reply:
done